### ISSUE TITLE
mSEED file read as tar file and hangs

### ISSUE THREAD

**Issue by Jollyfant** [obspy/obspy#2288]
Hi, there is a problem with the following mSEED file:

http://www.orfeus-eu.org/documents/IU.SFJD.00.VMZ.D.2014.295

There seems to be a little coincidence that some magic bytes match that of a tar archive. When reading the file ObsPy checks and delegates it to an upstream library. Both Python 2 and Python 3 report that this is a tar file, but it hangs forever on Python 2.7.

    >>> import tarfile
    >>> tarfile.is_tarfile("IU.SFJD.00.VMZ.D.2014.295)"
    True

The line in ObsPy:

https://github.com/obspy/obspy/blob/master/obspy/core/util/decorator.py#L156

Any ideas on what we should do? Maybe we can fix this in ObsPy with one extra check.. but the hanging seems to be an upstream problem of `tarfile`.

**Comment by megies** [comment]
Hmm.. I guess we need a `kwarg` on `read()` that allows the user to force skipping the automatic decompress check..

**Comment by megies** [comment]
> When is support for Python 2 planned to be dropped anyway?

open for discussion.. https://github.com/obspy/obspy/issues/2295#issuecomment-457532797

**Comment by ThomasLecocq** [comment]
I can't reproduce this, for none of the two files, using the ``maintenance`` branch

one option is to manually pass ``check_compression`` = ``False`` to ``read``

**Comment by ThomasLecocq** [comment]
```
from obspy import read
print(read("http://www.orfeus-eu.org/documents/IU.SFJD.00.VMZ.D.2014.295"))
print(read("http://www.orfeus-eu.org/documents/NA.SEUT..SBT.D.2015.282"))
```
works ok for me

**Comment by megies** [comment]
Uhm, yeah @Jollyfant I also can read these files with `read()` on Py27 without problems..

**Comment by Jollyfant** [comment]
🤔 okay.. I gotta check the version of ObsPy on that server otherwise I'll close it as solved.

**Comment by Jollyfant** [comment]
It's actually 1.0.3 so I guess it was fixed in 1.1.0!

**Comment by megies** [comment]
> Hmm.. I guess we need a `kwarg` on `read()` that allows the user to force skipping the automatic decompress check..

Looks like this was fixed by #1436 but it didn't make it in the changelog, @krischer 


Btw.. we already have a kwarg to skip this, see #1845  :see_no_evil:

**Comment by krischer** [comment]
Lol 🙈 I knew this issue sounded familiar but I could not place my finger on it...
